tests/test_cli.py

```diff
@@ -136,6 +136,14 @@ def test_cli(run_cli, geojson_path):
 
     run_cli("--geometry", geojson_path, "-q", "producttype=GRD,polarisationmode=HH")
 
+@pytest.mark.fast
+def test_complex_geometry(run_cli, fixture_path):
+    with pytest.raises(ValueError):
+        run_cli(
+                '--geometry', fixture_path('map_complex_geometry.geojson'),
```

Comment:
We should probably not add a rather big GeoJSON file to this repo just for this test. Let us generate that dynamically in a fixture instead with some random coordinates. Ideally, this geometry never gets sent to the API endpoint anyways, since the validation you added kicks in before.